docs/source/blueprints/BP005.rst

```diff
@@ -0,0 +1,275 @@
+We ran a poll to choose the name and `autils` (Avocado Utils) was the winner.
+So our goal is to host common libraries under `avocado-project/autils`
+repository on GitHub and make them also available on PyPI under `python-autils`
+package name, since the `autils` name is unavailable.
+
+Utility onboarding
+==================
+
+The primary goal of autils is to be a central place for modules that ease the
+interaction with system level features and interfaces.  As such, if a user
+needs to interact with such a system level feature or interface, he/she may
+propose a new module.
+
+For instance, if a user finds himself/herself constantly interacting with disk
+partitions, he/she may want to propose a “partitions” module that provides
+basic APIs for that. In this case it makes sense to have it as part of this
+central repository.
+
+However, if the utility intends to ease the interaction with an esoteric
+feature, system or device, this may not be a good place for it.
+
+As a general rule, if a system level feature or interface is generically
+available to a multitude of users, such as the partition example above, it is a
+good candidate for this repository. Common sense is much appreciated here.
+
+Besides that, the proposed utility must have at least one MAINTAINER that will
+be responsible for keeping track of Pull Requests, tests and quality
+requirements as described below:
+
+Quality requirements
+--------------------
+
+It’s expected that both new modules and changes to existing modules will need
+to observe a full set of quality requirements and general standards that will
+either be described (documentation) and enforced as automated checks as much as
+possible.
+
+Among the topics related to the quality requirements and standards, the
+contribution documentation and automated checks will cover, at least:
+
+* Logical, stylistic, analytical and formatter checks
+
+  * Docstring checks
+  * Pylint
+  * PEP-8, PEP-257, etc…
+
+* Unit and Functional Tests Coverage
+
+  Due to the current code quality and given this is a new repository we would
+  like to reach at least 90% of coverage. This way, users writing tests relying
+  on these libraries will have a guarantee of a good level of correctness and
+  quality, redirecting the debugging and troubleshooting efforts away from the
+  well tested libraries.
+
+The new standalone repository will have predefined pipelines to run those
+checks, so new libraries will be automatically enforced to newly introduced
+libraries.
+
+Metadata description
+--------------------
+
+Every module will need to provide a description including:
+
+* Utility description
+* The utility category (ie: network, cpu, security, etc..)
+* Maintainers information
+  * Main contact (We need a contact point, a single maintainer or a group)
+  * Maintainer Group members (this will be used to block merges)
+* Supported Platforms
+* Supported execution method (Local vs Remote)
```

Comment:
What is meant by this? Is this somehow related to `aexpect`?

Reply:
It means that we intend to flag modules that have the capability of acting on remote systems.  This will probably be a broad classification, just for informational purposes because unfortunately we have way too many ways of executing code remotely.

The `avocado.utils.network` introduced a "remote" approach based on the definition of 
a base `Host` class, along with `LocalHost` and `RemoteHost`.  Some other libraries (`avocado.utils.distro` for instance) will allow the use of a `session` parameter.  And of course `aexpect` has a lot of the same (and more) functionality.

We don't hope to address a complete standardization of the remote execution APIs just yet, but we want to at least bring some visibility to it.

Reply:
Exactly that. Thanks @clebergnu .